### PR TITLE
[14.0][IMP] base_tier_validation: Misc changes

### PR DESCRIPTION
**Pull request by victoralmau**
Changes done:
- [x] Add `_check_state_from_condition()` method to better clarify the code and to be able to inherit it by another module.
- [x] Add `_add_tier_validation_x()` methods to better clarify the code and to be able to inherit it by another module.
- [x] Add name to `th` + `td` from ReviewsTable for a better selector if another module needs it.

@Tecnativa TT43351

### CODE REVIEW

File: base_tier_validation/static/src/xml/tier_review_template.xml [hunk 1]
```diff
@@ -32,7 +32,7 @@
                     <th class="text-right">Todo by</th>
                     <th class="text-right">Done by</th>
                     <th class="text-right">Validation Date</th>
-                    <th class="text-right">Comment</th>
+                    <th name="th_comment" class="text-right">Comment</th>
```
Comment (pedrobaeza) [inline base_tier_validation/static/src/xml/tier_review_template.xml:35]: Add this to all the th and contents.

**Comment by victoralmau** [comment]
Changes done.

**Review by pedrobaeza — APPROVED** [review]

**Review by CarlosRoca13 — APPROVED** [review]

**Comment by victoralmau** [comment]
It's ready to merge, right?

**Pull request merged**